model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -155,3 +156,9 @@ std::unique_ptr<ResultBase> SegmentationModel::postprocess(InferenceResult& infR
 
     return std::unique_ptr<ResultBase>(result);
 }
+
+std::unique_ptr<ResultBase> SegmentationModel::infer(const ImageInputData& inputData)
```

Comment:
It should return `ImageResult`: https://github.com/openvinotoolkit/open_model_zoo/blob/391af4e4f050da4a57a72bdb8d63498e51cccd67/demos/segmentation_demo/cpp/main.cpp#L386